src/trading/MakeOrder.sol

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controlled.sol';
+import 'ROOT/libraries/trading/PlaceOrder.sol';
+import 'ROOT/libraries/ReentrancyGuard.sol';
+import 'ROOT/trading/IMakeOrder.sol';
+import 'ROOT/trading/Trading.sol';
+
+
+contract MakeOrder is Controlled, ReentrancyGuard {
+    using PlaceOrder for PlaceOrder.Data;
+
+    function makeOrder(Trading.TradeTypes _type, uint256 _attoshares, int256 _displayPrice, IMarket _market, uint8 _outcome, bytes32 _betterOrderId, bytes32 _worseOrderId, uint256 _tradeGroupId) onlyInGoodTimes nonReentrant external returns (bytes32 _orderId) {
+        PlaceOrder.Data memory _placeOrderData = PlaceOrder.create(controller, msg.sender, _outcome, _type, _attoshares, _displayPrice, _market, _betterOrderId, _worseOrderId, _tradeGroupId);
+
+        uint256 _fxpMoneyEscrowed;
+        uint256 _fxpSharesEscrowed;
+        if (_placeOrderData.tradeType == Trading.TradeTypes.Ask) {
+            (_fxpMoneyEscrowed, _fxpSharesEscrowed) = PlaceOrder.placeAsk(_placeOrderData);
+        } else if (_placeOrderData.tradeType == Trading.TradeTypes.Bid) {
+            (_fxpMoneyEscrowed, _fxpSharesEscrowed) = PlaceOrder.placeBid(_placeOrderData);
+        }
+
+        _orderId = _placeOrderData.orders.getOrderId(_placeOrderData.tradeType, _placeOrderData.market, _placeOrderData.attoshares, _placeOrderData.displayPrice, _placeOrderData.sender, block.number, _placeOrderData.outcome, _fxpMoneyEscrowed, _fxpSharesEscrowed);
```

Comment:
Consider making this a method on the `Order` class as well, or better yet memoize `orderId` lazily.  Really we don't need the order ID until we call `saveOrder`, and we can make a method on `Order` that checks to see if `Order.orderId == 0` and if so it gets the `orderId` from the current data and if not uses whatever `orderId` is set to.  Then the `saveOrder` function can call into `orders.saveOrder` with the appropriate parameters.

With the above, this callsite would simplify to just `_orderData.saveOrder();` and then on the last line of this function we would just have `return _orderData.orderId();`

The goal with these changes is to make this function easy to read and we hide away all of the complex function calls (calls with lots of parameters) and switching logic inside the Order class functions.